data/config-schema/src/main/resources/net/corda/schema/configuration/p2p.gateway/1.0/corda.p2p.gateway.json

```
@@ -25,25 +25,33 @@
       "pattern": "^\\/[\\/.a-zA-Z0-9-]*$"
     },
     "sslConfig": {
-      "description": "Used for one-way TLS between Gateways.",
+      "description": "Used for TLS between Gateways.",
       "type": "object",
       "default": {},
       "properties": {
         "revocationCheck": {
           "type": "object",
           "default": {},
-          "description": "How to check for revocation of the server certificate.",
+          "description": "How to check for revocation of the TLS certificate.",
           "properties": {
             "mode": {
-            "oneOf": [
-              {"const": "SOFT_FAIL", "title": "The revocation check will succeed if the revocation status cannot be determined under certain circumstances. See java.security.cert.PKIXRevocationChecker.Option.SOFT_FAIL java doc for more details."},
-              {"const": "HARD_FAIL", "title": "Opposite of SOFT_FAIL - i.e. most rigorous check. Among other things, this check requires that CRL checking URL is available on every level of certificate chain."},
-              {"const": "OFF", "title": "Switch CRL check off."}
-            ],
-            "default": "SOFT_FAIL"
-           }
+              "oneOf": [
+                {"const": "SOFT_FAIL", "title": "The revocation check will succeed if the revocation status cannot be determined under certain circumstances. See java.security.cert.PKIXRevocationChecker.Option.SOFT_FAIL java doc for more details."},
+                {"const": "HARD_FAIL", "title": "Opposite of SOFT_FAIL - i.e. most rigorous check. Among other things, this check requires that CRL checking URL is available on every level of certificate chain."},
+                {"const": "OFF", "title": "Switch CRL check off."}
+              ],
+              "default": "SOFT_FAIL"
+            }
           }
         },
+        "tlsType": {
+          "description": "How to validate TLS certificates in the gateway. This setting MUST be the same on all Gateways that need to communicate via HTTPS.",
+          "oneOf": [
+            {"const": "ONE_WAY", "title": "Only validate the TLS certificates for the gateway server."},
+            {"const": "MUTUAL", "title": "Validate the TLS certificates for both the gateway server and client."}
+          ],
+          "default": "ONE_WAY"
+        },
```

```suggestion
        "tlsType": {
          "description": "What kind of TLS connections to establish between the gateways in our cluster and gateways in other clusters. This setting MUST be the same on all Gateways that need to communicate via HTTPS.",
          "oneOf": [
            {"const": "ONE_WAY", "title": "Establishes a regular TLS connection, where the server gateway will be authenticated."},
            {"const": "MUTUAL", "title": "Establishes a mutual TLS connection, where both the server and client gateway are authenticated."}
          ],
          "default": "ONE_WAY"
        },
```
Suggestion to make the phrasing a bit more generic towards TLS connections, not just the certificates.